tommath_private.h

```diff
@@ -33,17 +33,53 @@ extern "C" {
 #  define MP_FREE_BUFFER(mem, size)   MP_FREE((mem), (size))
 #  define MP_FREE_DIGITS(mem, digits) MP_FREE((mem), sizeof (mp_digit) * (digits))
 #else
-#  define MP_FREE_BUFFER(mem, size)   do { size_t fs_ = (size); void* fm_ = (mem); if (fm_) { MP_ZERO_BUFFER(fm_, fs_); MP_FREE(fm_, fs_); } } while (0)
-#  define MP_FREE_DIGITS(mem, digits) do { int fd_ = (digits); void* fm_ = (mem); if (fm_) { MP_ZERO_BUFFER(fm_, sizeof (mp_digit) * (size_t)fd_); MP_FREE(fm_, sizeof (mp_digit) * (size_t)fd_); } } while (0)
+#  define MP_FREE_BUFFER(mem, size)                     \
+do {                                                    \
+   size_t fs_ = (size);                                 \
+   void* fm_ = (mem);                                   \
+   if (fm_ != NULL) {                                   \
+      MP_ZERO_BUFFER(fm_, fs_);                         \
+      MP_FREE(fm_, fs_);                                \
+   }                                                    \
+} while (0)
+#  define MP_FREE_DIGITS(mem, digits)                   \
+do {                                                    \
+   int fd_ = (digits);                                  \
+   void* fm_ = (mem);                                   \
+   if (fm_ != NULL) {                                   \
+      MP_ZERO_BUFFER(fm_, sizeof(mp_digit) * (size_t)fd_); \
+      MP_FREE(fm_, sizeof(mp_digit) * (size_t)fd_);     \
```

Comment:
@nijtmans mentioned an unnecessary duplication of the multiplication in #258. Could we fix it here please such that we don't touch the same code multiple times?

Reply:
Nevermind, I have rebased by PRs #258 and #260 based on this PR. No need to do anything here.